lib/galaxy/tools/search/__init__.py

```diff
@@ -162,7 +163,7 @@ def search(self, q, tool_name_boost, tool_section_boost,
             return rval
         else:
             # Use asterisk Whoosh wildcard so e.g. 'bow' easily matches 'bowtie'
-            parsed_query = self.parser.parse(cleaned_query + '*')
+            parsed_query = self.parser.parse('*' + cleaned_query + '*')
```

Comment:
Exciting - I always search by this and this never works. Thanks so much.

Maybe use the short ID (it is unfortunately called ``tool.old_id``, not a great name) so you don't have to modify this? I would think you'd generally want to search on the short id anyway.

Reply:
I don't remember exactly why I removed this asterisk, but I think it was a combination of performance, false hits, and ngram support. See https://github.com/galaxyproject/galaxy/commit/d0eb6aba558e99f8461ae0677d6a54b545c0dc92 and https://github.com/galaxyproject/galaxy/pull/8789 for the change

> do not prepend a leading * wildcard for the individual terms, this should make things much faster on a big index, it will also yields less, but better results